src/sonic-config-engine/portconfig.py

```diff
@@ -51,4 +120,94 @@ def parse_port_config_file(port_config_file):
             port_alias_map[data['alias']] = name
     return (ports, port_alias_map)
 
+# Generate configs (i.e. alias, lanes, speed, index) for port
+def gen_port_config(ports, parent_intf_id, index, alias_at_lanes, lanes,num_lane_used, speed, assigned_lane,  offset):
```

Comment:
I think we can still cleanup this code by making use of i in the loop. If possible take care of it now, otherwise will do it as a followup while integrating the code

Reply:
I am not addressing this issue right now. As i's value is changing as per step, it's creating some complexity in the code.